src/drt/src/db/tech/frConstraint.h

```diff
@@ -1145,6 +1148,62 @@ class frLef58SpacingEndOfLineConstraint : public frConstraint
   std::shared_ptr<frLef58SpacingEndOfLineWithinConstraint> withinConstraint;
 };
 
+class frLef58SpacingWrongDirConstraint : public frConstraint
+{
+ public:
+  // constructors
+  frLef58SpacingWrongDirConstraint() = default;
+  // getters
+  frCoord getWrongDirSpace() const { return wrongDirSpace; }
+  bool hasNoneolWidth() const { return noneolValid; }
+  frCoord getNoneolWidth() const { return noneolWidth; }
+  bool hasLength() const { return lengthValid; }
+  frCoord getLength() const { return length; }
+  bool hasPrlLength() const { return prlLengthValid; }
+  frCoord getPrlLength() const { return prlLength; }
+  // setters
+  void setWrongDirSpace(frCoord wrongDirSpaceIn)
+  {
+    wrongDirSpace = wrongDirSpaceIn;
+  }
+  void setNonEol(bool noneolIn) { noneolValid = noneolIn; }
+  void setNonEolWidth(frCoord noneolWidthIn) { noneolWidth = noneolWidthIn; }
+  void setLengthValid(bool lengthValidIn) { lengthValid = lengthValidIn; }
+  void setLength(frCoord lengthIn) { length = lengthIn; }
+  void setPrlLengthValid(bool prlLengthValidIn)
+  {
+    prlLengthValid = prlLengthValidIn;
+  }
+  void setPrlLength(frCoord prlLengthIn) { prlLength = prlLengthIn; }
+  // others
+  frConstraintTypeEnum typeId() const override
+  {
+    return frConstraintTypeEnum::frcLef58SpacingWrongDirConstraint;
+  }
+  void report(utl::Logger* logger) const override
+  {
+    logger->report(
+        "SPACING WRONGDIRECTION wrongDirSpace {} noneolValid {} noneolWidth {} "
+        "lengthValid {} length {} prlLengthValid {} prlLength {} ",
+        wrongDirSpace,
+        noneolValid,
+        noneolWidth,
+        lengthValid,
+        length,
+        prlLengthValid,
+        prlLength);
+  }
+
+ protected:
+  frCoord wrongDirSpace{0};
+  bool noneolValid{false};
+  frCoord noneolWidth{0};
+  bool lengthValid{false};
+  frCoord length{0};
+  bool prlLengthValid{false};
+  frCoord prlLength{0};
```

Comment:
I see no reason for having both prlLengthValid and prlLength. if no PRL specified, then prlLength would be 0. Same for the odb object